app/views/search/_attachment.html.erb

```diff
@@ -1,11 +1,15 @@
-<div class="title">
-  Attachment on story <%= link_to model.story.name, [model.story.project, model.story] %> in <%= link_to model.story.project.name, model.story.project %>
-</div>
+<% if model.story.present? %>
```

Comment:
I've had to add this check across each of the result types - you can't add a `touch: true` to a `has_many` relation, and so when a story has been archived - it isn't updating the attachment/comment etc, and so the PG Search isn't indexing the change.

Another option would be to write our own method on `after_save` to touch any of the relations manually, and we could avoid this. What you think?